### DISCUSSION TITLE
writeQueueSize and udpMaxPayloadSize

### DISCUSSION THREAD

**Discussion by GyudongPark** [bluenviron/mediamtx#3660] [Questions]
### Question

How these two options can affect latency of WebRTC?

**Comment by aler9** [comment]
They do not.

Packets are sent out as soon as they are received and there are no internal buffers.

`writeQueueSize` affects the size of the queue of outgoing packets. It should be increased only if you get error "write queue is full".

`udpMaxPayloadSize` affects the size of outgoing UDP packets. It should not be touched since it's already set to the maximum possible size of an UDP packet (~1400 bytes), unless there are network restrictions that force you to decrease them further (for instance, when you are using a VPN, that might take some bytes from the packet).

Latency in WebRTC is unfortunately chosen by browsers by using an internal algorithm. You can provide a "hint" to this algorithm in order to set the desired latency. I posted an explanation here: https://github.com/bluenviron/mediamtx/issues/4063#issuecomment-2558071097
